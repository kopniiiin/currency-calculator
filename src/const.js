export const Currency = {
  EUR: `EUR`,
  USD: `USD`,
  JPY: `JPY`
};

export const currenciesToTranslations = {
  [Currency.EUR]: `ÐµÐ²Ñ€Ð¾`,
  [Currency.USD]: `Ð´Ð¾Ð»Ð»Ð°Ñ€`,
  [Currency.JPY]: `Ð¸ÐµÐ½Ð°`
};

export const currenciesToEmoji = {
  [Currency.EUR]: `ðŸ’¶`,
  [Currency.USD]: `ðŸ’µ`,
  [Currency.JPY]: `ðŸ’´`
};

export const RATE_PRECISION = 2;
