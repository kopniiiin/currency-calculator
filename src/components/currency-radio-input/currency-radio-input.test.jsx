import React from "react";
import {shallow} from "enzyme";

import {Currency} from "../../const.js";

import CurrencyRadioInput from "./currency-radio-input.jsx";

describe(`snapshot test: CurrencyRadioInput`, () => {
  it(`should render correctly`, () => expect(
    shallow(
      <CurrencyRadioInput
        currency={Currency.EUR}
        name={`currency`}
        value={Currency.EUR}
        label={`currency`}
        checked={false}
        onChange={() => {}}/>
    )
  ).toMatchSnapshot());

  it(`should render checked`, () => expect(
    shallow(
      <CurrencyRadioInput
        currency={Currency.EUR}
        name={`currency`}
        value={Currency.EUR}
        label={`currency`}
        checked={true}
        onChange={() => {}}/>
    )
  ).toMatchSnapshot());
});

describe(`e2e test: CurrencyRadioInput`, () => {
  it(`should call onChange with value`, () => {
    const value = Currency.EUR;
    const onChange = jest.fn();

    shallow(
      <CurrencyRadioInput
        currency={Currency.EUR}
        name={`currency`}
        value={value}
        label={`currency`}
        checked={false}
        onChange={onChange}/>
    ).find(`.currency-radio-input__input`).simulate(`change`, {target: {value}});

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange.mock.calls[0][0]).toBe(value);
  });
});
