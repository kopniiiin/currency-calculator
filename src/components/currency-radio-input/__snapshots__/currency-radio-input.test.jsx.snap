// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: CurrencyRadioInput should render checked 1`] = `
<label
  className=" currency-radio-input currency-radio-input_currency_EUR currency-radio-input_active"
>
  <input
    aria-label="currency"
    checked={true}
    className="currency-radio-input__input"
    name="currency"
    onChange={[Function]}
    type="radio"
    value="EUR"
  />
</label>
`;

exports[`snapshot test: CurrencyRadioInput should render correctly 1`] = `
<label
  className=" currency-radio-input currency-radio-input_currency_EUR "
>
  <input
    aria-label="currency"
    checked={false}
    className="currency-radio-input__input"
    name="currency"
    onChange={[Function]}
    type="radio"
    value="EUR"
  />
</label>
`;
